### ISSUE TITLE
support for quizzes

### ISSUE THREAD

**Issue by specu** [sealemar/BingRewards#351]
Looking at network traffic, it looks like the user gets credited via POST requests to:
'https://www.bing.com/rewardsapp/reportActivity' and 'https://www.bing.com/rewardsapp/reportActivity' every time a guess is clicked. Those are the same pages that service hits. First call always has the same parameters in one quiz. Second one is a more elaborate contains unique identifiers and I can't seem to locate all the parameters in the HTML page.

first call is, might be optional, or perhaps it provides parameters for the second call:
data = '{ActivitySubType : quiz, ActivityType : notification, OfferId : Gamification_DailySet_20180207_Child2, Channel : Bing.Com, PartnerId : BingTrivia}'
response = requests.post('https://www.bing.com/msrewards/api/v1/ReportActivity', headers=headers, cookies=cookies, data=data)

second call is:
params = (
    ('IG', 'unique identifier'),
    ('IID', 'SERP.5050'),
    ('q', 'What is the host city for the 2018 Winter Olympics?'),
    ('filters', 'BTQI:"0" BTCI:"0" ShowTimesTaskPaneTrigger:"false"'),
    ('skipopalnative', 'true'),
    ('rnoreward', '1'),
    ('FORM', 'ML151V'),
    ('BTCA', '1'),
    ('BTSPC', '0'),
)

data = [
  ('url', 'https://www.bing.com/search?q=What is the host city for the 2018 Winter Olympics%3F&filters=BTQI%3A%220%22 BTCI%3A%220%22 ShowTimesTaskPaneTrigger%3A%22false%22&skipopalnative=true&rnoreward=1&FORM=ML151V&BTCA=1&BTSPC=0'),
  ('V', 'web'),
]

response = requests.post('https://www.bing.com/rewardsapp/reportActivity', headers=headers, params=params, cookies=cookies, data=data)

If all those parameters can be harvested from the html, then i think it's worth a try. Obviously there should be a random wait time between calls. And preferably it would be great to avoid the requests module. I'll keep poking around.

**Comment by baltimoron** [comment]
I haven't tried this yet so I was wondering: can you just grab the URL itself from somewhere? If so then you wouldn't have to search for the individual params in the html.

**Comment by specu** [comment]
@baltimoron i dont think so - same as with new way the hits register.

**Comment by baltimoron** [comment]
Actually I just double checked and the params are hard-coded for that one so yeah it's different.

**Comment by baltimoron** [comment]
Only thing I'm having trouble with here is the IG param. It exists in the source but the value never matches the IG value in the request.

**Comment by specu** [comment]
there's actually one more call that's related to quizzes - ncsheader or something like that. There's a quiz tomorrow so I'll try to take a look.

**Comment by zengfu94** [comment]
so proud, ur use of curl.trillworks brings tears to my eyes
may everyone know of its greatness

ps u dont need second call ;) just forget about it

**Comment by baltimoron** [comment]
Hey @specu, any chance you've been able to get the first call to execute correctly? I'm getting a 500 response any time I try to include the data, and nothing if I don't. That call is only run when you click on the correct answer, so I'm thinking it's the most important. Also, if you look at the cookies, there's one called _SS that appears to be tracking quiz progress. I've tried manually manipulating it but that doesn't seem to help.

**Comment by specu** [comment]
@baltimoron no, not yet. I've been planning to experiment on the quiz from the email aka.ms/Feb_US_RNL_Quiz over the weekend (since it doesnt count towards streaks), I hope to have time for it.

**Comment by specu** [comment]
every day I'm hoping to look into this and the days are just flowing by.

**Comment by aliasfoxkde** [comment]
@specu I know the feeling.

**Comment by specu** [comment]
we should have a bounty set up for this and other. I'm sure there's enough people here and if everyone would pitch in measly $5, it might be worthwhile to someone with time.

**Comment by aliasfoxkde** [comment]
I'll gladly drop $20 on it if someone had the time to get it working. Need to be verified working though, but yeah definitely a useful feature. Interesting idea @specu. I've been making small improvements to my own version (like a bulk script through vpn, android/ipython support, and network binding but I haven't had nearly as much time as I've wanted on this to try and get weekly points or quizes working).

**Comment by zengfu94** [comment]
idk about a bounty and im not trying to advertise but my private bot does this and tons of other things with many happy customers. https://t.me/joinchat/C30ZkhCvWoPtAGWiTpmBpg

**Comment by specu** [comment]
@baltimoron I've tried to get them to successfully register a quiz answer by making just the first call to /msrewards/api/v1/ReportActivity as zengfu suggested but I kept getting error 500 even after copying most of the headers from live session (I only replaced user-agent to match what script used). I'm going to try replicate cookies but i think those would be automatically generated by calls to those 2 calls that happen prior to /v1/ReportActivity - meaning ncheader and /ReportActivity with no v1 (I'm typing from memory). Hopefully some of this works, but so far it looks like 500 is generate because of missing cookies.

**Comment by specu** [comment]
@baltimoron I've tried to get them to successfully register a quiz answer by making just the first call to /msrewards/api/v1/ReportActivity as zengfu suggested but I kept getting error 500 even after copying most of the headers from live session (I only replaced user-agent to match what script used). I'm going to try replicate cookies but i think those would be automatically generated by calls to those 2 calls that happen prior to /v1/ReportActivity - meaning ncheader and /ReportActivity with no v1 (I'm typing from memory). Hopefully some of this works, but so far it looks like 500 is generate because of missing cookies.

**Comment by rustyshackleford2017** [comment]
@baltimoron @specu I don't know much about code but this bot apparently could do the quizzes, even the drag and drop ones. Maybe you could borrow some of the code from his script:

https://github.com/kathawala/RewardsBot

It uses Selenium though, maybe that's why kathawala was able to implement it, because he was using Selenium to emulate a browser and this code is more lower level automation? Just a thought based on my very limited knowledge of scripting.

**Comment by specu** [comment]
that's a completely different approach. selenium basically enables moving mouse cursor in firefox using script. this script sends and receives http to bing.

**Comment by specu** [comment]
@baltimoron 
I'm trying to send info to https://www.bing.com/msrewards/api/v1/ReportActivity
so that it registers responses to quizes. Data sent is the same with every answer and it looks like the information about the progress is provided via _SS cookie when that call is made.
I can share what i got if you want to take a look. it gives me error 500 at the moment.
Have you had a chance to make any progress?

**Comment by lukedukeus** [comment]
I know this thread is kind of dead, but I am willing to offer a bounty for solving this one, https://github.com/sealemar/BingRewards/issues/381 .
I have already made a bot using urllib2 to scrape reddit and grab email links, then search them for points, I just cant find out where to implement it into the code. my code is posted on my page, and I am  offering a $5 bounty if anyone can show me it implemented. It would be easy, but i dont know how to code.
